include/networkit/centrality/TopHarmonicCloseness.hpp

```diff
@@ -8,160 +8,145 @@
+     * However, for most real-world networks the empirical running time is
+     * O(m).
+     *
+     * @param G The input graph. If useNBbound is set to true, edge weights will be ignored.
+     * @param k Number of nodes with highest harmonic closeness that have to be found.
+     * For example, k = 10, the top 10 nodes with highest harmonic closeness will be computed.
+     * @param useNBbound If true, the NBbound variation will be used, otherwise NBcut.
+     */
+    explicit TopHarmonicCloseness(const Graph &G, count k = 1, bool useNBbound = false);
+
+    ~TopHarmonicCloseness() override;
+
+    /**
+     * Computes top-k harmonic closeness on the graph passed in the constructor.
+     */
+    void run() override;
+
+    /**
+     * Returns a list with the k nodes with highest harmonic closeness.
+     * WARNING: closeness centrality of some nodes below the top-k could be equal
+     * to the k-th closeness, we call them trail. Set the parameter includeTrail
+     * to true to also include those nodes but consider that the resulting vector
+     * could be longer than k.
+     *
+     * @param includeTrail Whether or not to include trail nodes.
+     * @return The list of the top-k nodes.
+     */
+    std::vector<node> topkNodesList(bool includeTrail = false);
+
+    /**
+     * Returns a list with the scores of the k nodes with highest harmonic
+     * closeness.
+     * WARNING: closeness centrality of some nodes below the top-k could
+     * be equal to the k-th closeness, we call them trail. Set the parameter
+     * includeTrail to true to also include those centrality values but consider
+     * that the resulting vector could be longer than k.
+     *
+     * @param includeTrail Whether or not to include trail centrality value.
+     * @return The closeness centralities of the k most central nodes.
+     */
+    std::vector<edgeweight> topkScoresList(bool includeTrail = false);
+
+private:
+    const Graph *G;
+    const count k;
+    const bool useNBbound;
+
+    std::vector<double> hCloseness;
+    std::vector<count> reachableNodes;
+    std::vector<std::vector<uint8_t>> visitedGlobal;
+    std::vector<uint8_t> tsGlobal;
+
+    std::vector<node> topKNodes;
+    std::vector<double> topKScores;
+
+    // For NBbound
+    std::vector<count> reachU, reachL;
+    std::vector<std::vector<count>> numberOfNodesAtLevelGlobal;
+    std::vector<std::vector<node>> nodesAtCurrentLevelGlobal;
+    std::vector<std::vector<std::vector<node>>> nodesAtLevelGlobal;
+    std::vector<double> levelImprovement;
+    std::unique_ptr<WeaklyConnectedComponents> wccPtr;
+    void updateTimestamp() {
+        auto &visited = visitedGlobal[omp_get_thread_num()];
+        auto &ts = tsGlobal[omp_get_thread_num()];
+        if (ts++ == std::numeric_limits<uint8_t>::max()) {
+            ts = 1;
+            std::fill(visited.begin(), visited.end(), 0);
+        }
+    }
+
+    template <class Type>
```

Comment:
That's probably a good idea. Since these `Less` and `Greater` structs are also used in other parts of the codebase, what about doing that in another PR?

Reply:
Agreed